File: backend/src/main/scala/microservices/content/objects/UpdateReviewPlannerRequest.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UpdateReviewPlannerRequest(
    userId: String,
    reviewId: String,
    rating: Int,
    title: String,
    content: String,
    images: List[ContentImagePlannerResponse]
)
object UpdateReviewPlannerRequest:
  given sourceEncoder: Encoder[UpdateReviewPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[UpdateReviewPlannerRequest] = deriveDecoder

File: frontend/src/microservices/content/objects/UpdateReviewPlannerRequest.ts
// UpdateReviewPlannerRequest：content 域更新评论请求对象。

import type { ContentImagePlannerResponse } from './ContentImagePlannerResponse'

export type UpdateReviewPlannerRequest = {
  userId: string
  reviewId: string
  rating: number
  title: string
  content: string
  images: ContentImagePlannerResponse[]
}

export const updateReviewPlannerRequestFromJson = (json: string): UpdateReviewPlannerRequest =>
  JSON.parse(json) as UpdateReviewPlannerRequest

export const updateReviewPlannerRequestToJson = (value: UpdateReviewPlannerRequest): string =>
  JSON.stringify(value)
